sim/rogue/sinister_strike.go

```diff
@@ -48,13 +48,13 @@ func (rogue *Rogue) registerSinisterStrikeSpell() {
 			OutcomeApplier: rogue.OutcomeFuncMeleeSpecialHitAndCrit(rogue.MeleeCritMultiplier(true, true)),
 			OnSpellHitDealt: func(sim *core.Simulation, spell *core.Spell, spellEffect *core.SpellEffect) {
 				if spellEffect.Landed() {
-					points := 1
-					if rogue.HasMajorGlyph(proto.RogueMajorGlyph_GlyphOfSinisterStrike) {
+					points := int32(1)
+					if rogue.HasMajorGlyph(proto.RogueMajorGlyph_GlyphOfSinisterStrike) && spellEffect.DidCrit() {
```

Comment:
Could the `HasMajorGlyph` return value be cached outside the function instead? Doing it inside the callback like this is very slow.

Reply:
Done